stdlib/array.ncl

```diff
@@ -346,5 +408,200 @@
         if %length% arr <= 1
         then arr
         else [ %head% arr ] @ (%tail% arr |> flat_map (fun a => [v, a])),
+
+    slice
+      : forall a. Number -> Number -> Array a -> Array a
+      | doc m%"
+          Results in the slice of the given array between `start` (included) and
+          `end` (excluded).
```

Comment:
I'm not a big fan of the "Results in" formulation. It sounds unidiomatic to me. But I'm not a native English speaker, so take that with a grain of salt. While we don't have a `return` construct in Nickel, I feel it would be better to use `Returns` in the documentation.

A separate issue is that we don't have a way of naming parameters in type signatures (outside of using records). So when reading this first sentence, it's not actually clear which parameters `start` and `end` refer to. Similar situations occur a lot in the stdlib documentation and spelling out a call like in this suggestions seems to be the best solution.
```suggestion
          `slice start end array` returns the slice of `array` between `start` (included) and
          `end` (excluded).
```